### ISSUE TITLE
Subscript expressions

### ISSUE THREAD

**Issue by pepsiman** [andreasfertig/cppinsights#44]
https://en.cppreference.com/w/cpp/language/operator_member_access says
"The built-in subscript expression E1[E2] is exactly identical to the expression *(E1 + E2)"
It's a useful insight when E1 is not an array or pointer, but E2 is.

**Comment by andreasfertig** [comment]
Hello pepsiman,

wonderful. That's one of this nice language features where I ask myself why it is there. I did a quick test and if that the transformation is easy at a first glance, all cases are more difficult. Once again I'm thinking about an option to show it ore leave it the way it is. Of course, a PR is welcome.

**Comment by languagelawyer** [comment]
> The built-in subscript expression E1[E2] is exactly identical to the expression *(E1 + E2)

Not exactly since C++14.
https://timsong-cpp.github.io/cppwp/n4140/expr.sub : "except that in the case of an array operand, the result is an lvalue if that operand is an lvalue and an xvalue otherwise."

```
using int_arr_t = int[2];
int main()
{
    #define E1 int_arr_t{ 0, 2 }
    #define E2 1
    #if 0
    E1[E2] = 1; // ill-formed, assignment requires lvalue, not xvalue
    #else
    *((E1)+(E2)) = 1; // well-formed
    #endif
}
```

**Comment by andreasfertig** [comment]
@languagelawyer  thanks for pointing that out. I'm unaware of that change.

While reading my former response I think I should apologize for my bad English. Let me try again.
I did a quick test whether the requested transformation is easy to implement, which at the first glance is the case. But a bigger amount of tests does not compile with that change. Apart from that, C++ Insights does not have options, which adds another layer of todo.
I still keep this issue open and hope to get to it in near the future.

**Comment by andreasfertig** [comment]
Hello @pepsiman,

a patch for this issue is on its way. For more details please refer to #45.
